include/ignition/gui/qml/GzVector3.qml

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+import QtQuick 2.9
+import QtQuick.Controls.Material 2.1
+import QtQuick.Layouts 1.3
+
+/**
+ *  Item displaying a 3D vector
+ *
+ *  Users can set values to xValue, yValue, and zValue.
+ *  If readOnly == False,
+ *  users can read from signal parameters of GzVectorSet: _x, _y, and _z
+ *
+ *  Usage example:
+ *  GzVector3 {
+ *    id: gzVector
+ *    xName: "Red"
+ *    yName: "Green"
+ *    zName: "Blue"
+ *    gzUnit: ""
+ *    readOnly: false
+ *    xValue: xValueFromCPP
+ *    yValue: yValueFromCPP
+ *    zValue: zValueFromCPP
+ *    onGzVectorSet: {
+ *      myFunc(_x, _y, _z, _roll, _pitch, _yaw)
+ *    }
+ *  }
+**/
+Item {
+  id: gzVectorRoot
+
+  // Readn-only / write
+  property bool readOnly: true
+
+  // User input value
+  property double xValue
+  property double yValue
+  property double zValue
+
+  /**
+   * Used to read spinbox values
+   * @params: _x, _y, _z: corresponding spinBoxes values
+   * @note: When readOnly == false, user should read spinbox value from its
+   *        parameters.
+   *        When readOnly == true, this signal is unused.
+   */
+  signal gzVectorSet(double _x, double _y, double _z)
+
+  // Names for XYZ
+  property string xName: "X"
+  property string yName: "Y"
+  property string zName: "Z"
+
+  // Units, defaults to meters.
+  // Set to "" to omit the parentheses.
+  property string gzUnit: "m"
+
+  // Expand/Collapse of this widget
+  property bool expand: true
+
+
+  /*** The following are private variables: ***/
+  height: gzVectorContent.height
+
+  // Maximum spinbox value
+  property double spinMax: Number.MAX_VALUE
```

Comment:
This should be moved out of the private variables section since users can overwrite